pkg/apis/pipeline/v1/container_types.go

```diff
@@ -129,9 +129,7 @@ type Step struct {
 
 	// OnError defines the exiting behavior of a container on error
 	// can be set to [ continue | stopAndFail ]
-	// stopAndFail indicates exit the taskRun if the container exits with non-zero exit code
-	// continue indicates continue executing the rest of the steps irrespective of the container exit code
-	OnError string `json:"onError,omitempty"`
+	OnError OnErrorType `json:"onError,omitempty"`
```

Comment:
we added support for variables in addition to the two constants - https://github.com/tektoncd/pipeline/pull/5307

```
steps[].onError: $(params.CONTINUE)
steps[].onError: continue
steps[].onError: stopAndFail
```

Reply:
Hi @pritidesai , we will eventually parse the variables to be actual values, and we have validation for the variable scenario. Is there any concern with it?

Reply:
Hi @pritidesai, thanks for letting me know. Was wondering if there is any concern when using variables with "OnErrorType". We will eventually parse the variable to a real value here.

Reply:
How?

I haven't reviewed the code changes in this PR. I am not sure how can it be restricted to the type of `enum` (`continue` and `stopAndFail`) when it can have a parameter of type `string`.

Reply:
This is also what I found interesting that ```enum``` is not that restricted in Go, even with a Typed string ```OnErrorType``` you can still do something like

```
step.OnError = "this is undefined value in enum OnErrorType"
```

and there is no compilation or runtime error. So there is essentially no difference between a regular string or a typed string ```OnErrorType```when calling [ApplyReplacement function](https://github.com/tektoncd/pipeline/blob/7ba5090c37ddc09be4c7ff62cc49a2428af22dc7/pkg/container/step_replacements.go#L27) other than some type casting.

Reply:
thank you @QuanZhang-William for all the efforts but declaring `onError` to as `enum` but allowing it to specify variables sounds very confusing. If we are declaring a field as enum, it will be simpler and easier to follow if it's restricted to the specified enum values. 

I suggest keeping the type to `string`, thanks!

Reply:
@pritidesai I am not sure I follow here ? The `type` is still `string`, as `OnErrorType` is still just an alias to a string. There is no concept of `enum` in Go anyway 🙃. Also, this is the exact same thing with *a lot* of other types we are using (in status, …), it is mainly useful for developer and consumer of the go library to use a predefined set of values, it doesn't do much more than this.